replica/table.cc

```diff
@@ -3555,7 +3555,7 @@ future<row_locker::lock_holder> table::do_push_view_replica_updates(shared_ptr<d
     co_await utils::get_local_injector().inject("table_push_view_replica_updates_timeout", timeout);
     auto lock = co_await std::move(lockf);
     auto pk = dht::partition_range::make_singular(m.decorated_key());
-    auto permit = sem.make_tracking_only_permit(base, "push-view-updates-2", timeout, tr_state);
+    auto permit = co_await sem.obtain_permit(base, "push-view-updates-2", estimate_read_memory_cost(), timeout, tr_state);
```

Comment:
There is another spot above, see `push-view-updates-1`. I see the comments say no read before write, so maybe the permit is only used to .... what is it used for?

And BTW, maybe it is time we give these a better name than `-1` and `-2`, like `view-update-read-before-write` and `view-update-no-view-before-write`.

Reply:
> There is another spot above, see `push-view-updates-1`. I see the comments say no read before write, so maybe the permit is only used to .... what is it used for?

It's used for creating the reader from mutation (the view update itself)
It's probably less likely to use much memory then when performing an actual read but after reconsidering, I'll add the admission there as well to be safe

> And BTW, maybe it is time we give these a better name than `-1` and `-2`, like `view-update-read-before-write` and `view-update-no-view-before-write`.

Added in rebase

Reply:
If it only reads from memory, than a tracking permit is fine.

Reply:
Ok, reverted in rebase